src/extensions/resolveSource.ts

```diff
@@ -10,35 +14,85 @@ export interface RegistryEntry {
   icons?: { [key: string]: string };
   labels: { [key: string]: string };
   versions: { [key: string]: string };
+  updateWarnings?: { [key: string]: UpdateWarning[] };
+}
+
+export interface UpdateWarning {
+  from: string;
+  description: string;
+  action?: string;
 }
 
 /**
- * Gets the source for a given extension name and version from the official extensions registry
+ * Gets the sourceUrl for a given extension name and version from the official extensions registry
  *
- * @param extensionName the name, or name@version of the extension to get the ExtensionSource for
+ * @param name the name of the extension to get the ExtensionSource for
```

Comment:
totally optional, and I almost don't want to say anything, but it's now bothering me a lot and I can't help it:

can we not end all these parameter descriptions with prepositions? `@param name the name of the extension` would be sufficient, `the name of the extension to fetch the ExtensionSource` is maybe a little more verbose, but fine. `for` adds absolutely nothing.

(And, if you do trim some prepositions, can you have all of the lines end with periods? It's an odd hodge-podge of them... need the formatter to really do it, so I don't really worry about it ever)

Reply:
Happy to run thru and clean these up. However, to be pedantic, [there is nothing wrong with ending a sentence with a preposition](https://www.merriam-webster.com/words-at-play/prepositions-ending-a-sentence-with).